src/electos/ballotmaker/ballots/files.py

```diff
@@ -0,0 +1,63 @@
+# Finds the source directory for the specified package,
+# builds paths and confirms the existence of files
+
+from pathlib import Path
+
+
+class FileTools:
+    def __init__(
+        self, file_name="", rel_path="assets/img", package_name="ballotmaker"
+    ):
+        self.file_name: str = file_name
+        self.rel_path: Path = Path(rel_path)
+        self.ext: str = ""
+        self.file_found: bool = False
+        self.package_name: str = package_name
+        self.package_path: Path = ""
+
+        # get the absolute path to this module
+        self.code_dir = Path(__file__).parent.resolve()
+        # TODO: restore test!
+        # does the package name appear in the code path?
+        # if self.code_dir.match(self.package_name):
+        # walk the path upward until the source dir
+        path_finder = self.code_dir
+        while path_finder.name != self.package_name:
+            path_finder = path_finder.parent
```

Comment:
Suggestion:

```
  assert path_finder != "/", "Path search reached the file system root."
```

Reply:
I've rewritten this module to search for "ballotmaker" in a python repo that contains the word "ballotmaker" in the path that seems to resolve this issue with our current Actions test suite.

I've also filed a bug ticket, [Avoid infinite loop in path check · Issue #97 · TrustTheVote-Project/BallotLab](https://github.com/TrustTheVote-Project/BallotLab/issues/97).

Therefore, I'd like to put off the ultimate resolution of this issue, which may simply be replacing this code with [python/importlib_resources](https://github.com/python/importlib_resources), as you suggested.

Reply:
@ion-oset not resolved but added to Issues (#92)